pkg/sloop/storemanager/storemanager.go

```diff
@@ -244,11 +292,31 @@ func cleanUpTimeCondition(minPartition string, maxPartition string, timeLimit ti
 	return false
 }
 
-func cleanUpFileSizeCondition(stats *storeStats, sizeLimitBytes int) bool {
-	if stats.DiskSizeBytes > int64(sizeLimitBytes) {
+func cleanUpFileSizeCondition(stats *storeStats, sizeLimitBytes int, gcThreshold float64) (bool, float64) {
+
+	requiredSizeLimit := gcThreshold * float64(sizeLimitBytes)
+	currentDiskSize := float64(stats.DiskSizeBytes)
+	if currentDiskSize > requiredSizeLimit {
 		glog.Infof("Start cleaning up because current file size: %v exceeds file size: %v", stats.DiskSizeBytes, sizeLimitBytes)
-		return true
+
+		garbageCollectionRatio := (currentDiskSize - requiredSizeLimit) / currentDiskSize
+		return true, garbageCollectionRatio
 	}
+
 	glog.V(2).Infof("Can not clean up, disk size: %v is not exceeding size limit: %v yet", stats.DiskSizeBytes, uint64(sizeLimitBytes))
-	return false
+	return false, 0.0
+}
+
+func getNumberOfKeysToDelete(db badgerwrap.DB, garbageCollectionRatio float64) float64 {
```

Comment:
It was because type Gauge interface only had functions with float64. As some of number of keys were emitted as metrics so I had set them to float64. But it makes sense to have integer variables and convert in the end while emitting. I have addressed this